node/tests/SharedTests.ts

```diff
@@ -5579,6 +5580,180 @@ export function runBaseTests<Context>(config: {
         },
         config.timeout,
     );
+
+    it.each([ProtocolVersion.RESP2, ProtocolVersion.RESP3])(
+        `lcs %p`,
+        async (protocol) => {
+            await runTest(async (client: BaseClient, cluster) => {
+                if (cluster.checkIfServerVersionLessThan("7.0.0")) return;
+
+                const key1 = "{lcs}" + uuidv4();
+                const key2 = "{lcs}" + uuidv4();
+                const key3 = "{lcs}" + uuidv4();
+                const key4 = "{lcs}" + uuidv4();
+
+                // keys does not exist or is empty
+                checkSimple(await client.lcs(key1, key2)).toEqual("");
+                checkSimple(await client.lcsLen(key1, key2)).toEqual(0);
+                checkSimple(await client.lcsIdx(key1, key2)).toEqual(
+                    new Map<string, ReturnType>([
+                        ["matches", []],
+                        ["len", 0],
+                    ]),
+                );
+
+                // LCS with some strings
+                checkSimple(
+                    await client.mset({
+                        [key1]: "abcdefghijk",
+                        [key2]: "defjkjuighijk",
+                        [key3]: "123",
+                    }),
+                ).toEqual("OK");
+                checkSimple(await client.lcs(key1, key2)).toEqual("defghijk");
+                checkSimple(await client.lcsLen(key1, key2)).toEqual(8);
+
+                // LCS with only IDX
+                checkSimple(await client.lcsIdx(key1, key2)).toEqual(
```

Comment:
Since the result is a more complex object, we should maybe wait for https://github.com/valkey-io/valkey-glide/pull/2052 to be done so we can throroughly check the structure of the result.

Reply:
Luckily it works, but if I change the expected value to `{matches: 6108123502, len: 8}`, it will also pass.
I agree that we need a better checking.